### PR TITLE
Using a badge/id shows a badge/id overlay on the character

### PR DESCRIPTION
**Pull request by pgmzeta**
[feature][game objects][player actions][sprites]
## About the PR 
When using a badge in-hand or on another player, show a sprite of the badge held up by the character's hand. The fingers holding the badge are tinted to match the character's gloves, skin tone or limb color.

a 'whip' SFX plays on showing emags and dabbing licenses.

## Why's this needed? 
Makes flashing your badge at someone feel more interactive/responsive.

## Screenshots
Screenshot 2024-07-20 232027
Screenshot 2024-07-20 225526
image

## TODO
- [x] Close on move/drop/laydown/etc

### CODE REVIEW

**Comment by LaCumbiaDelCoronavirus** [comment]
so it actually wasnt a joke..
amazing

**Comment by LaCumbiaDelCoronavirus** [comment]
ok more seriously, might want to include the miner's concussive gauntlets and syndie combat gauntlets too

**Comment by Glamurio** [comment]
Also, maybe force a south dir when using when you're facing north, since it looks strange.
image


Outside of that, since you're refactoring, maybe tie the actual using of the badge to the cooldown, not just the overlay, to reduce spam.
image

**Comment by SirNooben** [comment]
should make it so that if they're right in your face when showing the badge it hits you and maybe disorients you a bit

**Comment by Glamurio** [comment]
Someone was mentioning this should need a sound effect, and I agree. Some kind of blinging sound, maybe along the lines of this one (or even outright this one):
https://pixabay.com/sound-effects/cartoon-wink-magic-sparkle-6896/

**Comment by Nexusuxen** [comment]
This has my full support, ESPECIALLY if you can flash your ID like this; but in either case this is goofy in such a good way.

File: code/mob/living/carbon/human.dm [hunk 1]
```diff
@@ -3623,3 +3623,77 @@
...
 				arrestState = ARREST_STATE_CONTRABAND
 	src.arrestIcon.icon_state = arrestState
+
+/mob/living/carbon/human/get_hand_color()
+	if (istype(src.gloves))
+		var/obj/item/clothing/gloves/gloves = src.gloves
+		if (gloves.color)
+			return gloves.color
+		else
+			switch(gloves.icon_state)
```
Comment (TobleroneSwordfish) [inline code/mob/living/carbon/human.dm:3633]: No, make these a var/proc on the gloves.

File: code/mob/living/carbon/human.dm [hunk 2]
```diff
@@ -3623,3 +3623,77 @@
...
+	var/obj/item/parts/limb = null
+	if(src.hand)
+		limb = src.limbs.l_arm
+	else
+		limb = src.limbs.r_arm
+	if (istype(limb))
+		if (limb.skintoned)
+			return limb.skin_tone
+		else
+			if (isplantlimb(limb))
```
Comment (TobleroneSwordfish) [inline code/mob/living/carbon/human.dm:3680]: Same here

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]

**Comment by TobleroneSwordfish** [comment]
Some serious feedback for a silly feature: it should definitely force you to face south when using it and probably should close when you move? Otherwise it looks very weird while you're walking around. Should also close when you drop the item, lie down etc.
The fingertip sprites could also do with being just a little bit thicker, as they're quite hard to see currently (especially with robotic hands)
image

File: code/obj/item/clothing/gloves.dm [hunk 3]
```diff
@@ -203,6 +203,48 @@ ABSTRACT_TYPE(/obj/item/clothing/gloves)
...
 		return 1
 
+	/// Fingertip color, used to tint overlays
+	proc/fingertip_color()
+		if (src.color)
+			return src.color
+		else
+			switch(src.icon_state)
+				if("black")
+					return "#535353"
```
Comment (ZeWaka) [inline code/obj/item/clothing/gloves.dm:213]: why isn't this a property of the gloves?

File: code/obj/item/mob_parts.dm [hunk 4]
```diff
@@ -378,6 +378,31 @@ ABSTRACT_TYPE(/obj/item/parts)
 	proc/on_life(datum/controller/process/mobs/parent)
 		return
 
+	/// Fingertip color, used to tint overlays
+	proc/fingertip_color()
+		if (src.skintoned)
+			return src.skin_tone
+		else
+			if (isplantlimb(src))
+				return "#3fb54f"
```
Comment (ZeWaka) [inline code/obj/item/mob_parts.dm:387]: Why isnt this a property of the limb?

File: code/obj/item/card.dm [hunk 5]
```diff
@@ -263,10 +266,34 @@ TYPEINFO(/obj/item/card/emag)
...
+	var/pixel_x_offset = 0
+	var/pixel_y_offset = 3
+	var/hand_icon_state = ""
+	if(user.hand)
+		hand_icon_state = "id_hold_l"
+		pixel_x_offset = 5
+	else
+		hand_icon_state = "id_hold_r"
+		pixel_x_offset = -5
+
```
Comment (ZeWaka) [inline code/obj/item/card.dm:285]: needs user dir south

**Comment by ZeWaka** [comment]
waaa

File: code/obj/item/clothing/gloves.dm (hunk 3 above)
Comment (pgmzeta) [reply code/obj/item/clothing/gloves.dm:213]: pain in the butt to double-add it for chameleon gloves but done

**Comment by pgmzeta** [comment]
okay that wasn't nearly as bad as a refactor as I was initially thinking

**Review by ZeWaka — APPROVED** [review]
It's too good not to merge

**Comment by AshCorr** [comment]
Picsart_24-08-02_18-34-44-990

**Pull request merged**